### ISSUE TITLE
Developer guide

### ISSUE THREAD

**Issue by atkinsonm** [rollerderby/scoreboard#660]
It would be nice if there was a contribution and developer guide for this project that could document:

* Recommended paths for development i.e. how to best set up your developer workspace for FE/BE development
* Guidelines for PRs, what should be included, what generally will/will not be accepted, maybe a pull request template

I struggled a bit with frontend development on Windows (e.g. Chrome dev tools not picking up my local workspace, exacerbated slightly when I tried to use WSL) enough to where a paragraph or two with some development/testing tips would have been helpful.

**Comment by frank-weinberg** [comment]
I agree that this would be nice to have. But I don't see myself having any meaningful resources for creating it in the foreseeable future.

**Comment by davidmreed** [comment]
I'd be interested in setting up a VS Code dev container-based contribution flow. Happy to contribute that if it sounds valuable.

**Comment by frank-weinberg** [comment]
As long as it doesn't interfere with my non-container VS Code setup (which it probably won't), I don't mind including it. I don't know if anybody would be using it, but it won't hurt to have it available.

**Comment by annamontare-edu** [comment]
@davidmreed I'm currently getting started developing for the scoreboard using dev containers, did you wind up trying a setup like that?

**Comment by annamontare-edu** [comment]
Started a "Contributing to CRG ScoreBoard" draft here: https://github.com/rollerderby/scoreboard/wiki/Contributing-to-CRG-ScoreBoard

This might be a bit more general than the original issue, but I figured it might be nice to have a general contributing guide as well as more detailed developer guidance.

**Comment by atkinsonm** [comment]
> Started a "Contributing to CRG ScoreBoard" draft here: https://github.com/rollerderby/scoreboard/wiki/Contributing-to-CRG-ScoreBoard
> 
> This might be a bit more general than the original issue, but I figured it might be nice to have a general contributing guide as well as more detailed developer guidance.

@aidanmontare-edu this is a good start. What do you think about putting that content in `.github/CONTRIBUTING.md`? GitHub provides [a few niceties](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) for using that pattern that might help with discovery versus the wiki (though a more detailed dev guide that references the contributing guidelines would also be welcome).

**Comment by annamontare-edu** [comment]
Yeah I like that idea! Whenever I next get time I can start on a developer guide as well. Since I’m just getting started with scoreboard development, I’m hoping I can take some notes that will be useful to others—Aidan MontareOn Jun 17, 2024, at 5:25 PM, Michael Atkinson ***@***.***> wrote:﻿

Started a "Contributing to CRG ScoreBoard" draft here: https://github.com/rollerderby/scoreboard/wiki/Contributing-to-CRG-ScoreBoard
This might be a bit more general than the original issue, but I figured it might be nice to have a general contributing guide as well as more detailed developer guidance.

@aidanmontare-edu this is a good start. What do you think about putting that content in .github/CONTRIBUTING.md? GitHub provides a few niceties for using that pattern that might help with discovery versus the wiki (though a more detailed dev guide that references the contributing guidelines would also be welcome).

**Comment by annamontare-edu** [comment]
@atkinsonm began a pull request, will write some more content soon!

**Comment by annamontare-edu** [comment]
And here's a page for a more detailed developer guide: https://github.com/rollerderby/scoreboard/wiki/Developer-Guide (again, will try to add some actual content soon)

**Comment by annamontare-edu** [comment]
@davidmreed put my beginning of a DevContainers setup in #715. I'm just editing on the command line right now, so I image you might want to add more tools to get a good IDE experience.

**Comment by frank-weinberg** [comment]
Just as a quick note from my side: I read over your drafts and agree with the approach you're taking so far. I'll look over the details and add my own inputs when I have more time. (Not before the fix for the groundhog day bug is at least PRed.)

**Comment by annamontare-edu** [comment]
Thanks, and much appreciated!
